docs/developers/tutorials/staking-contract.md

```diff
@@ -0,0 +1,635 @@
+
+Example of error:
+```
+error[E0554]: #![feature] may not be used on the stable release channel
+--> /home/user/elrondsdk/vendor-rust/registry/src/github.com-1ecc6299db9ec823/elrond-wasm-derive-0.33.0/src/lib.rs:4:12
+```
+
+## VSCode and rust-analyser extension
+
+VSCode: https://code.visualstudio.com/
+
+Assuming you're on Ubuntu, download the `.deb` version. Go to that folder:
+- open folder in terminal
+- run the following command: `sudo dpkg -i downloaded_file_name`
+
+rust-analyser: https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer
+Elrond VSCode extension: https://marketplace.visualstudio.com/items?itemName=Elrond.vscode-elrond-ide
+
+Both can be easily installed from the "Extensions" menu in VSCode.  
+
+# Creating the contract
+
+Run the following command in the folder in which you want your smart contract to be created:
+```
+erdpy contract new staking-contract --template empty
+```
+
+Open the generated folder, and you should have the following structure:  
+![img](/developers/staking-contract-tutorial-img/folder_structure.png)
+
+For now, comment all the code in the `empty_rust_test.rs` file (ctrl + "A", then ctrl + "/"). Otherwise, it will keep popping up errors as we modify the contract's code.
+
+# Setting up the workspace
+
+Now, to have all the extensions work properly, we have to setup our workspace. This is done by pressing `ctrl + shift + P` and selecting the "Elrond: Setup Workspace" option from the menu. Choose the "Yes" option on the pop-up menu.
+
+Now let's open the Elrond VSCode extension and try building our contract, to see if everything is properly set up. Go to the extension's tab, right-click on "staking-contract" and select the "Build Contract" option:  
+![img](/developers/staking-contract-tutorial-img/elrond_ide_extension.png)
+
+Alternatively, you can run `erdpy --verbose contract build` yourself from the VSCode terminal.
+
+After the building has completed, our folder should look like this:  
+![img](/developers/staking-contract-tutorial-img/folder_structure_2.png)
+
+A new folder, called `output` was created, which contains the compiled contract code. More on this is used later. For now, let's continue.
+
+# Your first lines of Rust
+
+Currently, we just have an empty contract. Not very useful, is it? So let's add some simple code for it. Since this is a staking contract, we'd expect to have a `stake` function, right?
+
+Let's add said function:
+```rust
+#[payable("EGLD")]
+#[endpoint]
+fn stake(&self) {}
+```
+
+Since we want this function to be callable by users, we have to annotate it with `#[endpoint]`. Also,since we want to be able to receive a payment, we mark it also as `#[payable("EGLD)]`. For now, we'll use EGLD as our staking token. Note that the contract does NOT need to be payable for it to receive payments on endpoint calls. The payable flag at contract level is only for receiving payments without endpoint invocation.
```

Comment:
I would highlight this info _Note that the contract does NOT need to be payable for it to receive payments on endpoint calls. The payable flag at contract level is only for receiving payments without endpoint invocation._

I think it works something like this:
~~~
Note that the contract does NOT need to be payable for it to receive payments on endpoint calls. The payable flag at contract level is only for receiving payments without endpoint invocation.
~~~

or something else

Reply:
:::note  
the contract does NOT need to be payable for it to receive payments on endpoint calls. The payable flag at contract level is only for receiving payments without endpoint invocation.